File: my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"
 
class RobotNewsStationNode : public rclcpp::Node 
{
public:
    RobotNewsStationNode(): Node("robot_news_Station"),  robot_name("Bender")
    {
        publisher = this->create_publisher<example_interfaces::msg::String>("robot_news",10);
        timer = this->create_wall_timer(std::chrono::milliseconds(500),
                                        std::bind(&RobotNewsStationNode::publishNews,this));

        RCLCPP_INFO(this->get_logger(), "Robot News Station iniciou!");
    }
 
private:
    void publishNews()
    {
        auto msg = example_interfaces::msg::String();
        msg.data = std::string("Olá, eu sou o " + std::string(robot_name)+ " da estação Robot News!");
        publisher->publish(msg);
    }

    std::string robot_name;
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher;
    rclcpp::TimerBase::SharedPtr timer;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStationNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String
 
 
class RobotNewsStationNode(Node): 
    def __init__(self):
        super().__init__("robot_news_station")
        self.declare_parameter("robot_name","Bender")

        self.robot_nome = self.get_parameter("robot_name").value
        self.publisher = self.create_publisher(String, "robot_news", 10)   
        self.timer     = self.create_timer(0.5, self.publish_news)
        self.get_logger().info("Robot News Station has been started")
    
    def publish_news(self):
        msg = String()
        msg.data = "Olá, eu sou " + str(self.robot_nome) + " da robot news station!"
        self.publisher.publish(msg)
        
 
def main(args=None):
    rclpy.init(args=args)
    node = RobotNewsStationNode()
    rclpy.spin(node)
    rclpy.shutdown()
 
 
if __name__ == "__main__":
    main()